src/OpenTelemetry.Instrumentation.Owin/Implementation/HttpInListener.cs

```diff
@@ -0,0 +1,34 @@
+// <copyright file="HttpInListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.Owin.Implementation
+{
+    internal class HttpInListener : ListenerHandler
```

Comment:
@denisivan0v, your middleware approach is pretty neat! My gut says you won't need this OpenTelemetry.Instrumentation.Owin project once you convert your instrumentation to use `ActivitySource` as suggested earlier. That is, by starting/stopping an Activity from your middleware using an `ActivitySource` allows the OpenTelemetry SDK to directly listen to the `ActivitySource`.